test/test_ssm.py

```diff
@@ -134,6 +142,199 @@ def test_ssm_init_non_dirq(self):
+        # make sure it was dealt with correctly
+        for dn in valid_dns:
+            # Capture the log output so we can use it in assertions
+            with LogCapture() as log:
+                message_valid = """APEL-summary-job-message: v0.2
+                        Site: RAL-LCG2
+                        Month: 3
+                        Year: 2010
+                        GlobalUserName: """ + dn + """
+                        VO: atlas
+                        VOGroup: /atlas
+                        VORole: Role=production
+                        WallDuration: 234256
+                        CpuDuration: 2345
+                        NumberOfJobs: 100
+                        %%"""
+
+                mock_handle_msg.return_value = message_valid, dn, None
+
+                ssm = Ssm2(self._brokers, self._msgdir, TEST_CERT_FILE,
+                            TEST_KEY_FILE, dest=self._dest, listen=self._listen,
+                            capath=self.ca_certpath)
+
+                ssm._save_msg_to_queue(message_valid, empaid)
+
+                print(str(log))
+
+                self.assertIn('Message saved to incoming queue', str(log))
+
+        # As there are several different ways messages can be rejected,
+        # Keep a count to test a different method for each dn
+        dnCount = 1
+        for dn in rejected_dns:
+            # Capture the log output so we can use it in assertions
+            with LogCapture() as log:
+                message_rejected = """APEL-summary-job-message: v0.2
+                        Site: RAL-LCG2
+                        Month: 3
+                        Year: 2010
+                        GlobalUserName: """ + dn + """
+                        VO: atlas
+                        VOGroup: /atlas
+                        VORole: Role=production
+                        WallDuration: 234256
+                        CpuDuration: 2345
+                        NumberOfJobs: 100
+                        %%"""
+
+                # Change the reason for method rejection for each dn
+                if dnCount == 1:
+                    # Pass no message, which will also need an error message
+                    mock_handle_msg.return_value = None, dn, "Empty text passed to _handle_msg"
+                elif dnCount == 2:
+                    # Pass an error message
+                    mock_handle_msg.return_value = message_rejected, dn, "Signer not in valid DNs list"
+                else:
+                    # Pass a different error message
+                    mock_handle_msg.return_value = message_rejected, dn, "Failed to verify message"
+
+                ssm = Ssm2(self._brokers, self._msgdir, TEST_CERT_FILE,
+                            TEST_KEY_FILE, dest=self._dest, listen=self._listen,
+                            capath=self.ca_certpath)
+
+                ssm._save_msg_to_queue(message_rejected, empaid)
+
+                print(str(log))
```

Comment:
There's this and the print below that also need removing.

Reply:
done